### PR TITLE
Pull request for my todo app

### PR DESCRIPTION
**Pull request by Katsiez**
Link to my deployed project: https://katas-todos.netlify.app/

My reflections on how this week's project turned out: Turned out pretty great I would say. Was fun to make!

Things I'd like to have clarified or explained in more detail: Maybe how to correctly implement moment.js but even without that, I think it went well.

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/png" href="pencil.png">
```
Comment (amandaflod) [inline code/public/index.html:7]: Cute favicon! ✏️

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+
+import { tasks } from "./reducers/tasks";
+
+import { TaskContent } from "./components/TaskContent";
+
+const reducer = combineReducers({
+  tasks: tasks.reducer,
+});
+
+const persistedState = localStorage.getItem("reduxState")
```
Comment (amandaflod) [inline code/src/App.js:13]: Great that you added a local storage! 🤩 This method seems very nice and simple!

File: code/src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <DatenTime>
+          <p>
+            {currentDate} - {currentTime}
+          </p>
+        </DatenTime>
+      </div>
+    </HeaderStyled>
+  );
+};
+export const HeaderStyled = styled.div`
```
Comment (amandaflod) [inline code/src/components/Header.js:23]: I think it is best practice to declare variables before you use them, which is why it is common to have your styled components before your component 😊

File: code/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import moment from "moment";
+
+import styled from "styled-components";
+
+export const Header = () => {
+  const currentDate = moment().format("dddd, MMM Do");
```
Comment (amandaflod) [inline code/src/components/Header.js:7]: Really nice addition with the date on the page!

File: code/src/components/ItemAdded.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+          type="text"
+          required
+          value={newTask}
+          placeholder="Do next . . .✏️"
+          onChange={(event) => setNewTask(event.target.value)}
+        />
+
+        <TaskButton
+          type="submit"
+          disabled={
```
Comment (amandaflod) [inline code/src/components/ItemAdded.js:40]: Another great detail that really makes the app feel super nice, well done! 🚀

File: code/src/components/ItemList.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { useSelector } from "react-redux";
+import { ItemsAll } from "components/ItemsAll";
+
+import styled from "styled-components";
+
+export const ItemList = () => {
+  const allTasks = useSelector((store) => store.tasks.list);
+  const wholeList = useSelector((store) => store.tasks.list.tasks);
+
+  if (wholeList.length === 0) {
```
Comment (amandaflod) [inline code/src/components/ItemList.js:12]: Also love that you added this for the empty list, looks great and the solution is very nice and simple 😍

File: code/src/components/ItemRemoved.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
+//ItemRemoved.js handles removal of todos/tasks
+
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import Swal from "sweetalert2";
```
Comment (amandaflod) [inline code/src/components/ItemRemoved.js:6]: great idea to add an alert here 👏

File: code/src/components/TaskContent.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+
+import { Header } from "./Header";
+import { ItemAdded } from "./ItemAdded";
+import { ItemList } from "./ItemList";
+import { ItemRemoved } from "./ItemRemoved";
+
+import styled from "styled-components";
+
+export const TaskContent = () => {
```
Comment (amandaflod) [inline code/src/components/TaskContent.js:10]: Nice that you added this component to add structure to the page, makes it really clear!

File: code/src/index.css [hunk 9]
```diff
@@ -1,13 +1,27 @@
 body {
```
Comment (amandaflod) [inline code/src/index.css:1]: Beautiful styling as always Kat!! 😍 Makes you happy to look at the app 😊

**Review by amandaflod — COMMENTED** [review]
As always I am super impressed with what you have done! 🤩 It is so inspiring to see the level of detail and all the different tools you have used, I love that you are never scared to try something new! Great work Kat ❤️🐯

File: code/src/components/ItemRemoved.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        onClick={removeItems}
+        disabled={clearTasks === 0}
+      >
+        <h5>
+          remove all tasks
+          <span role="img" aria-label="bin">
+            🗑️
+          </span>
+        </h5>
+      </Button>
```
Comment (HIPPIEKICK) [inline code/src/components/ItemRemoved.js:51]: You don't need the `h5` inside your button. Wouldn't be cool if this was not even rendered in the case of `clearTasks === 0` ? :)

File: code/src/reducers/tasks.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  initialState: initialState,
+
+  reducers: {
+    taskAdded: (state, action) => {
+      const { taskInfo } = action.payload;
+      state.list.tasks.push({ ...taskInfo, createdAt: moment() });
+      //state.list.tasks.push({id:Date.now(), name: taskInfo, createdAt:moment()})
+    },
+
+    taskRemoved: (state, action) => {
```
Comment (HIPPIEKICK) [inline code/src/reducers/tasks.js:72]: Maybe "removeTask" instead?

File: code/src/reducers/tasks.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    ],
+  },
+};
+
+export const tasks = createSlice({
+  name: "tasks",
+  initialState: initialState,
+
+  reducers: {
+    taskAdded: (state, action) => {
```
Comment (HIPPIEKICK) [inline code/src/reducers/tasks.js:66]: Maybe "addTask"?

**Pull request closed without merging**